### PR TITLE
Allow to pin/unpin articles in automatic lists

### PR DESCRIPTION
**Pull request by takeit**
| Q                         | A
| ------------------------- | ---
| Bug fix?                  | no
| New feature?              | yes
| BC breaks?                | yes
| Deprecations?             | no
| Tests pass?               | yes
| Changelog update required | yes
| Fixed tickets             | SWP-414
| License                   | AGPLv3

TODO:
- [x] - functional tests
- [x] - docs

### CODE REVIEW

File: src/SWP/Bundle/ContentListBundle/Resources/config/doctrine-orm/ContentList.orm.yml [hunk 1]
```diff
@@ -46,3 +46,9 @@ SWP\Component\ContentList\Model\ContentList:
         deletedAt:
             type: datetime
             nullable: true
+    oneToMany:
+        items:
+            targetEntity: SWP\Component\ContentList\Model\ContentListItemInterface
+            mappedBy: contentList
+            orphanRemoval: true
+            cascade: ["all"]
```
Comment (takeit) [inline src/SWP/Bundle/ContentListBundle/Resources/config/doctrine-orm/ContentList.orm.yml]: missing new line

File: src/SWP/Bundle/ContentListBundle/Form/Type/BooleanType.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ */
+
+namespace SWP\Bundle\ContentListBundle\Form\Type;
+
+use SWP\Bundle\ContentListBundle\Form\DataTransformer\BooleanStringToBooleanTransformer;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class BooleanType extends AbstractType
```
Comment (takeit) [inline src/SWP/Bundle/ContentListBundle/Form/Type/BooleanType.php]: move to storage bundle?

File: src/SWP/Bundle/ContentListBundle/Form/DataTransformer/BooleanStringToBooleanTransformer.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * @copyright 2016 Sourcefabric z.ú
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\Bundle\ContentListBundle\Form\DataTransformer;
+
+use Symfony\Component\Form\DataTransformerInterface;
+
+class BooleanStringToBooleanTransformer implements DataTransformerInterface
```
Comment (takeit) [inline src/SWP/Bundle/ContentListBundle/Form/DataTransformer/BooleanStringToBooleanTransformer.php]: move to storage bundle?

File: src/SWP/Bundle/ContentListBundle/Resources/config/serializer/Model.ContentList.yml [hunk 4]
```diff
@@ -23,7 +23,7 @@ SWP\Component\ContentList\Model\ContentList:
             expose: true
             type: string
         items:
-            expose: true
+            expose: false
```
Comment (takeit) [inline src/SWP/Bundle/ContentListBundle/Resources/config/serializer/Model.ContentList.yml]: remove it at all

File: src/SWP/Bundle/CoreBundle/Model/ContentListItem.php [hunk 5]
```diff
@@ -0,0 +1,9 @@
+<?php
+
```
Comment (takeit) [inline src/SWP/Bundle/CoreBundle/Model/ContentListItem.php]: add copyright

File: src/SWP/Bundle/CoreBundle/Model/ContentListItemInterface.php [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace SWP\Bundle\CoreBundle\Model;
```
Comment (takeit) [inline src/SWP/Bundle/CoreBundle/Model/ContentListItemInterface.php]: add copyright

File: src/SWP/Bundle/CoreBundle/DependencyInjection/SWPCoreExtension.php [hunk 7]
```diff
@@ -68,6 +69,9 @@ public function prepend(ContainerBuilder $container)
                         'content_list' => [
                             'model' => ContentList::class,
                         ],
+                        'content_list_item' => [
```
Comment (takeit) [inline src/SWP/Bundle/CoreBundle/DependencyInjection/SWPCoreExtension.php]: this prepended config can be moved to config file in CoreBundle instead

File: src/SWP/Bundle/ContentListBundle/Form/DataTransformer/BooleanStringToBooleanTransformer.php (hunk 3 above)
Comment (ahilles107) [inline src/SWP/Bundle/ContentListBundle/Form/DataTransformer/BooleanStringToBooleanTransformer.php]: or maybe common?

**Pull request merged**